frontend/src/features/Train/features/Image/components/ImageFlow.tsx

```diff
@@ -0,0 +1,313 @@
+import HtmlTooltip from "@/common/components/HtmlTooltip";
+import InfoIcon from "@mui/icons-material/Info";
+import {
+  Button,
+  Card,
+  Divider,
+  Paper,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import ReactFlow, {
+  Background,
+  BackgroundVariant,
+  Controls,
+  Edge,
+  Handle,
+  Node,
+  NodeTypes,
+  Position,
+  addEdge,
+  useEdgesState,
+  useNodesState,
+} from "reactflow";
+import "reactflow/dist/style.css";
+import {
+  ALL_LAYERS,
+  DEFAULT_LAYERS,
+  STEP_SETTINGS,
+} from "../constants/imageConstants";
+import { ParameterData } from "../types/imageTypes";
+import assert from "assert";
+import { nanoid } from "nanoid/non-secure";
+import { toast } from "react-toastify";
+
+interface ImageFlowProps {
+  setLayers?: (layers: ParameterData["layers"]) => void;
+}
+
+export default function ImageFlow(props: ImageFlowProps) {
+  const { setLayers } = props;
+
+  const initialNodes: Node<LayerNodeData>[] = [
+    ROOT_NODE,
+    ...DEFAULT_LAYERS.MNIST.map((layer, i) => ({
+      id: `${layer.value}-${i}`,
+      type: "textUpdater",
+      position: {
+        x: DEFAULT_X_POSITION,
+        y: (i + 1) * 100,
+      },
+      data: {
+        label: STEP_SETTINGS.PARAMETERS.layers[layer.value].label,
+        value: layer.value,
+        parameters: layer.parameters.slice(),
+        onChange: onChange,
+      },
+    })),
+  ];
+  const initialEdges: Edge[] = createInitialEdges();
+
+  const [nodes, setNodes, onNodesChange] = useNodesState(initialNodes);
+  const [edges, setEdges, onEdgesChange] = useEdgesState(initialEdges);
+
+  useEffect(() => {
+    if (setLayers) {
+      const layers = handleExportLayers();
```

Comment:
could this theoretically be slow if we had a humongous model?

non-blocking, but a question

Reply:
Theoretically, yes. The way how it works is the layers state get updated at every node or edge change. It is an expensive operation but so far, it's non-significant for the number of layers we have rn